sdk/storage/storage-internal-avro/src/AvroReadableFromStream.ts

```diff
@@ -55,9 +55,12 @@ export class AvroReadableFromStream extends AvroReadable {
           let chunk = this._readable.read(size);
           if (chunk) {
             this._position += chunk.length;
-            // chunk.lenght maybe less than desired size if the stream ends.
+            // chunk.length maybe less than desired size if the stream ends.
             resolve(this.toUint8Array(chunk));
             this._readable.removeListener("readable", callback);
+            this._readable.removeListener("error", reject);
```

Comment:
https://nodejs.org/api/events.html#events_emitter_once_eventname_listener

according to the doc one-time listener will be removed so we shouldn't need to remove by ourselves?

Reply:
>The next time eventName is triggered, this listener is removed and then invoked.
But these listener will only be removed if triggered.
Adding this because getting "max listener limit exceeded" warning.

Also, do you know how to guarantee the input stream (not fully consumed) be properly garbage-collected after the `AvroReadableFromStream` is unreachable. I assume it will as long as there is no more data consumer.